e2e/test/iothub/AuthenticationWithTokenRefreshDisposalTests.cs

```diff
@@ -0,0 +1,252 @@
+
+                await deviceClient.OpenAsync().ConfigureAwait(false);
+                deviceClients.Add(deviceClient);
+            }
+
+            // Close and dispose client instance 1.
+            // The closed client should report a status of "disabled" while the rest of them should be connected.
+
+            await deviceClients[0].CloseAsync().ConfigureAwait(false);
+            deviceClients[0].Dispose();
+
+            amqpConnectionStatuses[0].LastConnectionStatus.Should().Be(ConnectionStatus.Disabled);
+
+            Logger.Trace($"{nameof(ReuseAuthenticationMethod_MuxedDevices)}: Confirming the rest of the multiplexed devices are online and operational.");
+
+            bool notRecovered = true;
+            var sw = Stopwatch.StartNew();
+            while (notRecovered && sw.Elapsed < MaxWaitTime)
+            {
+                notRecovered = false;
+                for (int i = 1; i < devicesCount; i++)
+                {
+                    if (amqpConnectionStatuses[i].LastConnectionStatus != ConnectionStatus.Connected)
+                    {
+                        await Task.Delay(TimeSpan.FromSeconds(1)).ConfigureAwait(false);
+                        notRecovered = true;
+                        break;
+                    }
+                }
+            }
+
+            notRecovered.Should().BeFalse();
+
+            // Send a message through the rest of the multiplexed client instances.
+            for (int i = 1; i < devicesCount; i++)
+            {
+                await deviceClients[i].SendEventAsync(new Client.Message()).ConfigureAwait(false);
+                Logger.Trace($"Test with client {i} completed.");
+            }
+
+            // Close and dispose all of the client instances.
+            for (int i = 1; i < devicesCount; i++)
+            {
+                await deviceClients[i].CloseAsync().ConfigureAwait(false);
+                deviceClients[i].Dispose();
+            }
+
+            deviceClients.Clear();
+            amqpConnectionStatuses.Clear();
+
+            // Initialize the client instances by reusing the created authentication methods and open the connection.
+            for (int i = 0; i < devicesCount; i++)
+            {
+                DeviceClient deviceClient = DeviceClient.Create(testDevices[i].IoTHubHostName, authenticationMethods[i], new ITransportSettings[] { amqpTransportSettings });
+
+                var amqpConnectionStatusChange = new AmqpConnectionStatusChange(testDevices[i].Id, Logger);
+                deviceClient.SetConnectionStatusChangesHandler(amqpConnectionStatusChange.ConnectionStatusChangesHandler);
+                amqpConnectionStatuses.Add(amqpConnectionStatusChange);
+
+                await deviceClient.OpenAsync().ConfigureAwait(false);
+                deviceClients.Add(deviceClient);
+            }
+
+            // Ensure that all clients are connected successfully, and the close and dispose the instances.
```

Comment:
Comment could be more specific - Reused x509 cert provided for second instance successfully.

Reply:
Well, the goal here is to ensure that the client is connected => the tcp connection is established. All the resources that this tcp connection depends on will thus inherently need to be in proper state. I don't want to call out a specific resource here, since that may be misleading in the sense that it is not the only resource whose state is important. Does that make sense?

Reply:
Sure but also good to call out the point about reusing the certs wherever you are reusing it in the tests.